aten/src/ATen/native/nestedtensor/NestedTensorImpl.cpp

```diff
@@ -0,0 +1,110 @@
+#include <ATen/ATen.h>
+#include <ATen/NamedTensorUtils.h>
+#include <ATen/WrapDimUtils.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/nestedtensor/NestedTensorImpl.h>
+#include <c10/core/DispatchKey.h>
+
+namespace at {
+namespace native {
+
+NestedTensorImpl::NestedTensorImpl(
+    at::Tensor buffer,
+    at::Tensor nested_size_tensor)
+    : TensorImpl(
+          c10::DispatchKeySet({DispatchKey::NestedTensor}),
```

Comment:
If you want a CPU nested tensor to properly report `is_cpu` you also need to include the CPU/CUDA dispatch key as necessary. (Or I guess you can wait until @bdhirsh lands https://github.com/pytorch/pytorch/pull/72827 and then cartesian product nested with cpu/cuda, which is probably a good idea anyway)